src/agent/onefuzz/src/machine_id.rs

```diff
@@ -112,6 +116,42 @@ pub async fn get_os_machine_id() -> Result<Uuid> {
     Ok(Uuid::parse_str(&guid)?)
 }
 
+#[cfg(target_os = "macos")]
+pub async fn get_os_machine_id() -> Result<Uuid> {
+    // getting the machine ID can be done through mach APIs, but for
+    // simplicity, this uses a method originally used in
+    // `cpp_client_telemetry`.
+    // ref: https://github.com/microsoft/cpp_client_telemetry/commit/41315d1b77dcf7a419332b96ea4c7cab43760853
+
+    let output = Command::new("ioreg")
+        .args(&["-d2", "-c", "IOPlatformExpertDevice"])
+        .stdin(Stdio::null())
+        .stderr(Stdio::null())
+        .output()
+        .await
+        .context("ioreg failed to run")?;
+
+    let stdout = String::from_utf8_lossy(&output.stdout);
+
+    for line in stdout.split('\n') {
+        // "IOPlatformUUID" = "GUID_HERE"
+        if line.contains("IOPlatformUUID") {
```

Comment:
I suggest pulling this into a helper function, unit testing.

Reply:
The `target_os` specific `get_os_machine_id` is unit tested already.  Does the underlying parsing need something more specific?

https://github.com/microsoft/onefuzz/blob/b5388d360c956f69e4b49923acdb6ebc9c772716/src/agent/onefuzz/src/machine_id.rs#L142-L145

Reply:
Addressed in other comment.